lib/taiko.js

```diff
@@ -346,33 +352,45 @@ module.exports.client = () => clientProxy;
  * Allows switching between tabs using URL or page title.
  *
  * @example
- * await switchTo('https://taiko.gauge.org/') # switch using URL
- * await switchTo('Taiko') # switch using Title
+ * # switch using URL
+ * await switchTo('https://taiko.gauge.org/')
+ * # switch using Title
+ * await switchTo('Taiko')
+ * # switch using regex URL
+ * await switchTo(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/)
+ * # switch using regex Title
+ * await switchTo(/Go*gle/)
  *
  * @param {string} targetUrl - URL/Page title of the tab to switch.
  *
  * @returns {Promise}
  */
+
 module.exports.switchTo = async targetUrl => {
   validate();
-  let t = typeof targetUrl;
-  if (t !== 'string') {
+  if (
```

Comment:
Keeping this check, because i saw a test case which verifies if switchTo() throws a TypeError if no argument is passed. Hence not touching it. If i had directly converted it into RegExp, then switchTo() would evaluate to switchTo(/*/) and this will just switch to the current tab.